### ISSUE TITLE
TemplatedDictionary stores values in __dict__, leads to surprising behavior

### ISSUE THREAD

**Issue by hroncok** [xsuchy/templated-dictionary#4]
```pycon
>>> from templated_dictionary import TemplatedDictionary
>>> td = TemplatedDictionary()
>>> td.copy()
<templated_dictionary.TemplatedDictionary object at 0x7f8e5573ade0>, TemplatedDictionary({'_aliases': {}})
>>> td['copy'] = True  # a fictional configuration option
>>> td.copy()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'bool' object is not callable
```

Is this expected?

**Comment by praiskup** [comment]
I don't think anyone has analyzed so far whether this is or isn't expected :-) what is your opinion?

**Comment by hroncok** [comment]
It's certainly a surprising way to shoot yourself in the foot. If the attribute access to dictionary keys is desired, it can be implemented more safely.

**Comment by praiskup** [comment]
Seems like the cause is that we inherit from MutableMapping and yet we do:
https://github.com/xsuchy/templated-dictionary/blob/8bf2431a0ec198b9db73441963d8ebe5db3e4024/templated_dictionary/__init__.py#L59-L60

**Comment by praiskup** [comment]
Well, I am not sure if we need the ability to access items like `td.item`, instead of just `td["item"]` .. I bet that the latter is just OK..

**Comment by hroncok** [comment]
Not doing this will cause a backward incompatibility in existing mock configs.

**Comment by praiskup** [comment]
Indeed, that would be a similar change to bcd90f0dafa365575c4b101e6f5d98c4ef4e4b69. But I think that usage as you describe, in Mock configs, is rare, and thus we can afford to do it. WDYT?

**Comment by hroncok** [comment]
I generally dislike removing existing API without a deprecation period.
